### PR TITLE
Adding Props to `Slider` Component

### PR DESCRIPTION
**Pull request by marashad001**
## Changes - #1368 

Adding the ability for every DOM node to have a custom className passed to it

## Testing 

After adding the new props into `Slider.tsx`, a specific prop can be passed into the return() code for `Slider` that includes the new custom className.

## Docs

N/A

### CODE REVIEW

File: packages/itwinui-react/src/core/Slider/Slider.tsx [hunk 1]
```diff
@@ -152,6 +152,26 @@ export type SliderProps = {
...
+   */
+  sliderMinProps?: React.ComponentProps<'span'>;
+  /**
+   * Allows props to be passed for slider-max
+   */
+  sliderMaxProps?: React.ComponentProps<'span'>;
+  /**
+   * Allows props to be passed for slider-container
+   */
+  sliderContainerProps?: React.ComponentProps<'div'>;
```
Comment (mayank99) [inline packages/itwinui-react/src/core/Slider/Slider.tsx:166]: there is already `railContainerProps`, so `sliderContainerProps` is not needed

File: packages/itwinui-react/src/core/Slider/Slider.tsx [hunk 2]
```diff
@@ -152,6 +152,26 @@ export type SliderProps = {
    * Additional props for container `<div>` that hold the slider rail, thumbs, and tracks.
    */
   railContainerProps?: React.HTMLAttributes<HTMLDivElement>;
+  /**
+   * Allows props to be passed for slider-min
+   */
+  sliderMinProps?: React.ComponentProps<'span'>;
```
Comment (mayank99) [inline packages/itwinui-react/src/core/Slider/Slider.tsx:158]: lets remove `slider` prefix from all of these to make the names consistent

File: packages/itwinui-react/src/core/Slider/Slider.tsx [hunk 3]
```diff
@@ -490,6 +529,7 @@ export const Slider = React.forwardRef((props, ref) => {
           );
         })}
         <Track
+          {...sliderTrackProps}
```
Comment (mayank99) [inline packages/itwinui-react/src/core/Slider/Slider.tsx:532]: should go at the end. only `className` prop (if present) should go after it

File: packages/itwinui-react/src/core/Slider/Slider.tsx
Comment (mayank99) [inline packages/itwinui-react/src/core/Slider/Slider.tsx:1]: missing props for ticks area

Comment (marashad001) [reply packages/itwinui-react/src/core/Slider/Slider.tsx:1]: I'm on it

File: packages/itwinui-react/src/core/Slider/Slider.tsx (hunk 1 above)
Comment (marashad001) [reply packages/itwinui-react/src/core/Slider/Slider.tsx]: fixed

File: packages/itwinui-react/src/core/Slider/Slider.tsx (hunk 2 above)
Comment (marashad001) [reply packages/itwinui-react/src/core/Slider/Slider.tsx]: fixed

File: packages/itwinui-react/src/core/Slider/Slider.tsx (hunk 3 above)
Comment (marashad001) [reply packages/itwinui-react/src/core/Slider/Slider.tsx]: fixed

File: packages/itwinui-react/src/core/Slider/Slider.tsx (hunk 1 above)
Comment (mayank99) [reply packages/itwinui-react/src/core/Slider/Slider.tsx]: you still need to fix the existing `railContainerProps` to actually merge the classes correctly

File: packages/itwinui-react/src/core/Slider/Slider.tsx [hunk 4]
```diff
@@ -412,9 +437,18 @@ export const Slider = React.forwardRef((props, ref) => {
...
+          className={cx('iui-slider-ticks', tickProps?.className)}
+        >
           {tickLabels.map((label, index) => (
-            <Box as='span' key={index} className='iui-slider-tick'>
+            <Box
+              as='span'
+              {...tickProps}
+              key={index}
+              className='iui-slider-tick'
+            >
```
Comment (LostABike) [inline packages/itwinui-react/src/core/Slider/Slider.tsx:451]: `iui-slider-ticks` seem to be a wrapper class of `iui-slider-tick` (notice one has "s") so I think they need different props

Comment (LostABike) [reply packages/itwinui-react/src/core/Slider/Slider.tsx:451]: also don't forget to add test for this too

**Review by mayank99 — APPROVED** [review]
LGTM. lets merge it without second approval, as Annie is with a different team now

File: packages/itwinui-react/src/core/Slider/Slider.tsx (hunk 1 above)
Comment (marashad001) [reply packages/itwinui-react/src/core/Slider/Slider.tsx]: fixed

**Pull request merged**